Textile/Classes/Services/TTEAppStateLifecycleManager.m

```diff
@@ -0,0 +1,144 @@
+#import "TTEAppStateLifecycleManager.h"
+#import "Callback.h"
+
+typedef NS_CLOSED_ENUM(NSInteger, AppState) {
+  AppStateNone,
+  AppStateBackground,
+  AppStateForeground
+};
+
+@interface TTEAppStateLifecycleManager ()
+
+@property (nonatomic, weak) Textile *textile;
+@property (nonatomic, assign) AppState currentAppState;
+@property (nonatomic, strong) NSTimer *timer;
+@property (nonatomic, assign) UIBackgroundTaskIdentifier bgTaskId;
+
+@end
+
+@implementation TTEAppStateLifecycleManager
```

Comment:
This implementation of `TTELifecycleManager` uses app state events to start and stop the node. It's start and stop methods can also be called externally by calling `Textile.start` and `Textile.stopWithCompletion`.